src/core/components/fields/LabeledPasswordField.test.tsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * @vitest-environment jsdom
+ */
+
+import { expect, test } from "vitest"
+import { fireEvent, render, screen } from "test/utils"
+import { Form, FormProps } from "src/core/components/fields/Form"
+import { z } from "zod"
+import { LabeledPassWordFieldProps, LabeledPasswordField } from "./LabeledPasswordField"
+import { useState } from "react"
+
+interface MyFormProps<S extends z.ZodType<any, any>> extends FormProps<S> {
+  name: string
+  placeholder: string
+  label: string
+  inputType?: "text" | "password" | "email" | "number" | "textarea"
+  onChangeCallback: (val) => void
+}
+
+const placeholder = "Container placeholder"
+const name = "input 1"
+const label = "input_label"
+
+function MyForm<S extends z.ZodType<any, any>>(props: MyFormProps<S>) {
+  const { name, placeholder, inputType, onChangeCallback, label, ...formProps } = props
+
+  const [currType, setType] = useState(inputType)
+  const handlePasswordToggle = () => {
+    if (currType === "password") {
+      setType("text")
+    } else {
+      setType("password")
+    }
+  }
+
+  return (
+    <div>
+      <Form<S> {...formProps}>
```

Comment:
```suggestion
      <Form {...formProps}>
```

Don't think it is necessary to have this (also not problematic to keep it) – your call 😊